docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -734,6 +734,29 @@ class SharedService {
 <2> `@Lock(Lock.Type.READ)` overrides the value specified at class level. It means that any number of clients can invoke the method concurrently, unless the bean instance is locked by `@Lock(Lock.Type.WRITE)`. 
 <3> You can also specify the "wait time". If it's not possible to acquire the lock in the given time a `LockException` is thrown.
 
+=== Repeatable interceptor bindings
+
+Quarkus supports `@Repeatable` interceptor binding annotations, if they are declared on methods.
+For example:
+
+[source,java]
+----
+@ApplicationScoped
+class AllEncompassingService {
+  @Cached(cacheName = "foo") // <1>
+  @Cached(cacheName = "bar")
+  BigData heavyComputation() {
+    // ...
+  }
+}
+----
+<1> We have an imaginary interceptor that stores data to a cache.
+The corresponding interceptor binding is called `@Cached` and is declared as `@Repeatable`.
+In this example, the interceptor will store the method's return value in two caches, because we add `@Cached` twice.
+
+Repeatable interceptor bindings declared on classes and stereotypes are not supported, because there are some open questions around interactions with the Interceptors specification.
```

Comment:
Maybe we should clarify that repeatable interceptor bindings _declared on interceptor classes_ are supported?

Reply:
So this can get misleading pretty quickly IMHO. You always need to put an interceptor binding on two places:

- on the interceptor class;
- on the place that should be intercepted.

I don't know if there's a good name for the 2nd item, because that's what we would need here.

I don't think we should say anything like "Repeatable interceptor bindings declared on classes are not supported, unless the class is an interceptor class", because that conflates the two concepts.

Reply:
>     * on the interceptor class;

-> **Declaring Interceptor Bindings of an Interceptor**

>     * on the place that should be intercepted.

-> **Binding an Interceptor to a Component**

This is the spec wording. So I think that we support "Repeatable interceptor bindings declared on methods of component classes and interceptor classes"...

Reply:
Gotcha, I'll try to figure something out.

Reply:
WDYT about this?

```adoc
=== Repeatable interceptor bindings

Quarkus has a limited support for `@Repeatable` interceptor binding annotations.

When binding an interceptor to a component, you can declare multiple `@Repeatable` annotations on methods.
Repeatable interceptor bindings declared on classes and stereotypes are not supported, because there are some open questions around interactions with the Interceptors specification.
This might be added in the future.

As an example, suppose we have an interceptor that stores method's return values into a cache.
The corresponding interceptor binding would be called `@Cached` and would be declared as `@Repeatable`.
If we wanted to store the return value into two caches at the same time, we would add `@Cached` twice:

[source,java]
----
@ApplicationScoped
class AllEncompassingService {
  @Cached(cacheName = "foo")
  @Cached(cacheName = "bar")
  BigData heavyComputation() {
    // ...
  }
}
----

This is how interceptors are used.
What about creating an interceptor?

When declaring interceptor bindings of an interceptor, you can add multiple `@Repeatable` annotations to the interceptor class as usual.
This is useless when the annotation members are `@Nonbinding`, as would be the case for the `@Cached` annotation, but is important otherwise.

For example, suppose we have an interceptor that can automatically log method invocations to certain targets.
The interceptor binding annotation `@Logged` would have a member called `target`, which specifies where to store the log.
Our implementation could be restricted to console logging and file logging:

[source,java]
----
@Interceptor
@Logged(target = "console")
@Logged(target = "file")
class NaiveLoggingInterceptor {
  // ...
}
----

Other interceptors could be provided to log method invocations to different targets.
```